core/src/main/java/com/scalar/db/storage/jdbc/JdbcService.java

```diff
@@ -88,10 +89,17 @@ public Scanner getScanner(Scan scan, Connection connection)
 
     TableMetadata tableMetadata = tableMetadataManager.getTableMetadata(scan);
 
-    SelectQuery selectQuery =
-        scan instanceof ScanAll
-            ? buildSelectQueryForScanAll((ScanAll) scan, tableMetadata)
-            : buildSelectQueryForScan(scan, tableMetadata);
+    SelectQuery selectQuery;
```

Comment:
I think we need the same change for the `scan()` method to support the relational scan for JDBC transactions. Do you plan to do this in another PR?

Reply:
Yes, I will add it in another PR, including tests for it.